mmv1/api/resource.rb

```diff
@@ -435,6 +439,34 @@ def decoder?
       !@transport&.decoder.nil?
     end
 
+    def add_labels_related_fields(props)
+      props.each do |p|
+        if p.is_a? Api::Type::KeyValueLabels
+          props << build_effective_field('labels', p.field_min_version)
+        elsif p.is_a? Api::Type::KeyValueAnnotations
+          props << build_effective_field('annotations', p.field_min_version)
+        elsif (p.is_a? Api::Type::NestedObject) && !p.all_properties.nil?
+          p.properties = add_labels_related_fields(p.all_properties)
+        end
+      end
+      props
+    end
+
+    def build_effective_field(name, min_version)
+      description = "All of #{name} (key/value pairs)\
+ present on the resource in GCP, including the #{name} configured through Terraform,\
+ other clients and services."
+
+      Api::Type::KeyValuePairs.new(
```

Comment:
Are we able to use keyword arguments instead of positional ones? That would obviate the need for comments: https://medium.com/@sologoubalex/keyword-parameters-in-ruby-e93d7a268151

Reply:
Yeah,  `keyword arguments` is a better option.